pkg/agent.go

```diff
@@ -2,56 +2,96 @@ package pkg
 
 import (
 	"fmt"
-	"github.com/vmware-labs/marketplace-cli/v2/internal/models"
 	"io"
 	"log"
 	"net/http"
 	"net/url"
+
+	uuid2 "github.com/google/uuid"
+	"github.com/vmware-labs/marketplace-cli/v2/internal/models"
 )
 
 func (m *Marketplace) GetActiveOnPremInstance() {
 
 }
 
-func (m *Marketplace) AgentLogin(agentLoginReq *models.AgentLoginRequest) (*models.AgentLoginResponseRest, error) {
-	agentLoginRes := &models.AgentLoginResponseRest{}
-	log.Println("@@@@@@ Agent login initiated @@@@@@@@@@")
-	requestURL := MakeURL(
-		m.GetHost(),
-		fmt.Sprintf("/api/v1/on-prem/agent/auth"),
-		url.Values{},
-	)
-
-	log.Println("@@ requestRUL :@@ ", requestURL)
-
-	resp, err := m.Client.PostJSON(requestURL, agentLoginReq)
-	if err != nil {
-		log.Println("failed to perform agent login with error", err)
-
-		return agentLoginRes, fmt.Errorf("agent login failed")
-	}
-
-	if resp.StatusCode != http.StatusOK {
-		body, err := io.ReadAll(resp.Body)
+func (m *Marketplace) AgentLogin() error {
+	// if agent is not registered with Marketplace , generate new ID and register
+	if m.AgentConfig.CLIID == "" {
+		uuid, err := uuid2.NewUUID()
 		if err != nil {
-			return agentLoginRes, fmt.Errorf("failed to perform agent login %s : (%d)", agentLoginReq, resp.StatusCode)
+			return err
 		}
 
-		log.Println("response body :", body)
-
-		return agentLoginRes, fmt.Errorf("failed to perform agent login %s : (%d)", agentLoginReq, resp.StatusCode)
+		log.Println("Agent not registered with Marketplace. Generating new reference ID for agent")
+		m.AgentConfig.CLIID = uuid.String()
+		m.AgentConfig.Flush() // push the registered UUID to config for further interactions with Marketplace
 	}
 
-	err = m.DecodeJson(resp.Body, agentLoginRes)
+	// call marketplace API for login
+	response, err := m.AgentOperations.AgentLogin(&models.AgentLoginRequest{
+		UserID:         "rsharmila",
```

Comment:
from where we will get this userId , have we thought about it ?

Reply:
as I told earlier, the oAuth app doesn't have a user id, so until we get some native identifiers from on prem, we had decided to hardcode

Reply:
ok